pages/mesosphere/dcos/2.1/security/ent/tls-ssl/ar-custom/index.md

```diff
@@ -0,0 +1,242 @@
+
+## <a name="manually-placing-custom"></a>Manually placing custom external certificate
+
+
+The custom external certificate and the associated private key must be put in the `$DCOS_INSTALL_DIR/genconf/` directory on the bootstrap node:
+
+```bash
+cd $DCOS_INSTALL_DIR
+ls genconf/
+```
+```bash
+external-certificate.crt
+external-certificate-key.key
+```
+
+## <a name="manually-placing-master"></a>Manually placing the private key 
+
+
+For security reasons, the installer will not copy the private key from the bootstrap node to the master nodes.
+The private key associated with the custom external certificate must be distributed manually to every DC/OS master node **before starting the installation**.
+
+The filesystem path for the private key file must be `/var/lib/dcos/pki/tls/private/adminrouter-external.key`.
+The directory `/var/lib/dcos/pki/tls/private/` can be created manually with the following command before putting the file `adminrouter-external.key` in the directory on every DC/OS master node:
+
+```bash
+ mkdir -p /var/lib/dcos/pki/tls/private/
+```
+
+Furthermore, the file containing the private key `adminrouter-external.key` corresponding to the custom external certificate must be owned by the root Unix user and have 0600 permissions set.
+
+If you copy the private key file over the network onto the master nodes, the network channel must be adequately protected. An example of copying the external certificate private key is given below. The commands are executed on the bootstrap node. The `W.X.Y.Z` below indicates the IP address of a master node:
+
+```bash
+cd $DCOS_INSTALL_DIR/genconf
+scp external-certificate-key.key centos@W.X.Y.Z:/var/lib/dcos/pki/tls/private/adminrouter-external.key
+```
+
+## Specifying locations
+
+The filesystem paths to the custom external certificate and associated private key in the `$DCOS_INSTALL_DIR/genconf/` directory on the bootstrap node must be specified in the DC/OS configuration file using, respectively, the `external_certificate_path` and `external_certificate_key_path` parameters. The paths must be relative to `$DCOS_INSTALL_DIR`.
+
+The [Example use cases](#example-use-cases) section below shows how to set these configuration parameters.
+
+# <a name="config-ref"></a>Configuration parameter reference
+## external_certificate_path
+Path (relative to the `$DCOS_INSTALL_DIR`) to a file containing a single X.509 leaf certificate in the OpenSSL PEM format. For example: `genconf/external-certificate.crt`. If necessary, it should include all the intermediate CA certificates up to the root CA certificate trusted by the clients. It is advised to not to include the final root CA certificate in the PEM file, but is not strictly necessary.
+
+If provided, this is the custom external certificate.  If not provided, the DC/OS cluster generates a unique leaf certificate during the initial bootstrap phase signed by DC/OS CA and presents it to the clients.
```

Comment:
```suggestion
If provided, this is the custom external certificate.  If not provided, the DC/OS cluster uses a unique leaf certificate generated during the initial bootstrap phase signed by DC/OS CA and presents it to the clients.
```

Reply:
The AR cert signed by DC/OS CA is always generated, only when CAC is present then the CAC is used instead while communicating with external clients.